cpp/src/join/join_kernels.cuh

```diff
@@ -185,6 +185,48 @@ __global__ void compute_join_output_size(multimap_type multi_map,
   if (threadIdx.x == 0) atomicAdd(output_size, block_counter);
 }
 
+/* --------------------------------------------------------------------------*/
+/**
+* @brief  Computes the output size of joining the left table to the right table
+  by iterating over both and counting the number of matches.
+*
+* @param[in] left The left table
+* @param[in] right The right table
+* @param[in] check_row_equality The row equality comparator
+* @param[in] left_num_rows The number of rows in the left table
+* @param[in] right_num_rows The number of rows in the right table
+* @param[out] output_size The resulting output size
+  @tparam JoinKind The type of join to be performed
+*
+*/
+/* ----------------------------------------------------------------------------*/
```

Comment:
We've been removing dashed lines _inside_ doc comments, but that doesn't seem to be the case here since these are separate comment blocks. Does doxygen completely ignore these standalone comments? Does having a standalone comment between the doc comment and the corresponding function cause any issues? If this doesn't cause any issues with doxygen, it's fine with me, just asking since I'm unsure.

Reply:
I don't think the `detail` API docs are rendered, so I removed these dashes for consistency with the style guide.